matrix-org/synapse
# Carry IdP Session IDs through user-mapping sessions.

Since https://github.com/matrix-org/synapse/pull/11482, we're saving sessions IDs from upstream IdPs, but we've been loosing them when the user goes through a user mapping session on account registration.

This ensures the `auth_provider_session_id` gets carried through user mapping sessions.
How to test this: 

 - Spin up a Keycloak instance, create a realm
 - Configure Synapse to login via Keycloak: https://matrix-org.github.io/synapse/latest/openid.html#keycloak
 - Create a user in Keycloak
 - Try to login in Synapse. You should get the "choose your MXID" page
 - After completing the login, the `device_auth_providers` should have a row with the session ID (Keycloak uses UUIDs for those), which was not the case before


### Pull Request Checklist



* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))
